### ISSUE TITLE
Hide systemd output when booting live disk unless there is an error

### ISSUE THREAD

**Issue by leviport** [pop-os/iso#304]
**Distribution (run `cat /etc/os-release`):**
 ```
NAME="Pop!_OS"
VERSION="22.04 LTS"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 22.04 LTS"
VERSION_ID="22.04"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=jammy
UBUNTU_CODENAME=jammy
LOGO=distributor-logo-pop-os
```
 
 
**Related Application and/or Package Version (run `apt policy $PACKAGE NAME`):**
Probably systemd

**Issue/Bug Description:**
When booting the live disk, lots of terminal output is shown on the screen. 

**Steps to reproduce (if you know):**
Boot the live disk

**Expected behavior:**
Hide the terminal output unless something goes wrong with starting the graphical environment. The text on the screen should be the fallback, not the default.

**Other Notes:**

**Comment by leviport** [comment]
It wasn't a coincidence, lol. I opened this per your feedback.

Part of the reason for its current behavior is that Linux is historically booted/installed on all kinds of hardware, with varying levels of support. If booting the live disk failed, the user could see what the problem was and take action on that. Nowadays, most Linux distributions boot extremely reliably, so this text is usually not needed. 

When helping newer users troubleshoot, it's also very helpful to know where exactly booting failed. The person helping simply has to ask for a picture of their screen, or for the user to describe the last couple messages that were printed. The user can also start plugging some of those terms into a search engine and find some answers. If this output is hidden, or the user has to perform some extra step to make it appear, they're probably less likely to have quick answers. 

That's why I was sure to specify that this output should still appear when there is some kind of problem during booting. Hopefully that is doable. If it is not, I'd much rather leave the live disk boot process as it is now, even if some newer users are surprised to see it.

**Comment by Mesum-Hussain** [comment]
I see, are you the one managing the Pop OS Twitter account or are you somehow related to System 76?
I thought someone opened this issue coincidentally the same time as I was discussing it on Twitter 😅.

**Comment by Mesum-Hussain** [comment]
> That's why I was sure to specify that this output should still appear when there is some kind of problem during booting. Hopefully that is doable. If it is not, I'd much rather leave the live disk boot process as it is now, even if some newer users are surprised to see it.

Yeah you are right. That output might be helpful in case of any failure. 
But still I don't think a new user will think of posting these errors on forums for troubleshooting.
What people are used to on Windows and Mac OS is that they are informed that some error occurred without knowing any technicalities of the error. I don't personally think it's a good idea to adopt that on Linux, but we have to keep in mind that people are used to that. I think at least a text promoting the user to copy the output and search for the solution online, should be provided.
Maybe a suggesting r/pop_os would be a good idea.

**Comment by leviport** [comment]
>I see, are you the one managing the Pop OS Twitter account or are you somehow related to System 76?

I work for System76, but I don't manage the Twitter account. I like to keep an eye out for tweets about Pop!\_OS and System76. 

> But still I don't think a new user will think of posting these errors on forums for troubleshooting.

Not sure about that. I've helped lots of people troubleshoot booting problems, and they almost always give at least a small description of what they see on the screen.

>What people are used to on Windows and Mac OS is that they are informed that some error occurred without knowing any technicalities of the error.

Agreed, and I'd argue that we should avoid that at all costs. We can make things more user-friendly without robbing the user of any opportunity to see the inner workings of stuff. Transparent and slightly messy is much better than clean, but opaque.

>I think at least a text promoting the user to copy the output and search for the solution online, should be provided.

I bet once there's a mechanism for detecting when the boot process failed to start X11 (or Wayland, once that's the thing), there will be mechanisms for displaying instructions or tips alongside the error output. That would definitely be helpful.

**Comment by Mesum-Hussain** [comment]
> I work for System76, but I don't manage the Twitter account. I like to keep an eye out for tweets about Pop!\_OS and System76. 

👍🏻

> >What people are used to on Windows and Mac OS is that they are informed that some error occurred without knowing any technicalities of the error.
> 
> Agreed, and I'd argue that we should avoid that at all costs. We can make things more user-friendly without robbing the user of any opportunity to see the inner workings of stuff. Transparent and slightly messy is much better than clean, but opaque.


A new user actually might want to avoid such an opportunity to see the inner workings (I mean maybe that's why they choose a beginner friendly distro like Pop OS right?..... otherwise they would have gone with Arch or something).
Personally speaking, I was even scared to go into the BIOS to do things like disabiling Secure Boot (because according to the internet, it was there to protect my PC.....was thinking why does Linux requires disabling it?), then just imagine, the first thing you see after disabling Secure Boot and plugging in the USB is a running wall of text that you have no idea about, as if you injected a virus or your PC got hacked or something. 
It's a lot better to hide messy stuff from the beginners if they don't want to see it, and provide them with the option if they want to see it.
Beginners don't really try to troubleshoot stuff you know, they simply ditch the whole thing if it's not working.
Can say this because I literally did the same thing: my first choice was actually Zorin OS, but after failing at a few attempts of installing it, I decided to go with a different distro and that's how I ended up on Pop OS. Luckily I didn't decide to ditch Linux entirely.

Might be fine though, if you are targeting only the developers who have the habit to troubleshoot. But I really don't think that the goal of Linux should be only to make itself easy and accessible to the developers. We should strive for accessibility for everyone.


> >I think at least a text promoting the user to copy the output and search for the solution online, should be provided.
> 
> I bet once there's a mechanism for detecting when the boot process failed to start X11 (or Wayland, once that's the thing), there will be mechanisms for displaying instructions or tips alongside the error output. That would definitely be helpful.

Was that a jab at me? 🤔
Simple explanation would have been enough actually.
I mean I am not an OS developer, I don't know what technical issues are there while implementing this request. I am only suggesting what I think will make the experience more polished.

Anyways you do as you see fit. I was just suggesting this because I didn't had a good experience as an absolute beginner who was installing an OS for the first time. Just had good intentions to improve the experience of others.

**Comment by leviport** [comment]
>Was that a jab at me? :thinking:

...no? I'm failing to see how that statement could even be interpreted as a jab.

I'm also not sure why it sounds like you're still trying to make a case for something that we already both agree on, and we opened this issue specifically to track.

**Comment by Mesum-Hussain** [comment]
> >Was that a jab at me? :thinking:
> 
> ...no? I'm failing to see how that statement could even be interpreted as a jab.

Ok all good then I guess.

> I'm also not sure why it sounds like you're still trying to make a case for something that we already both agree on, and we opened this issue specifically to track.

Yeah maybe you are right. Sorry for stating my case when it's not necessary as we both already agree.
I was basically trying to say that even in case of an error, a new user is less likely to troubleshoot, so hiding the output even then, until the user themself opt to see it, looks more reasonable to me. 

A friend dialogue something on the lines of

*Oops! An error occurred 🥺.

Would you like to see the error log to troubleshoot?

Try taking a picture of it and posting it on r/pop_os subreddit for help.
*

**Comment by DivannKokos** [comment]
i hope that if ever hiding would be done, it would be done as an option and not the only way, cause imho i like wall of text more than any animation
wish if windows had wall of text hahahh

**Comment by leviport** [comment]
Everything is optional in Linux :wink:

**Comment by Mesum-Hussain** [comment]
> i hope that if ever hiding would be done, it would be done as an option and not the only way, cause imho i like wall of text more than any animation wish if windows had wall of text hahahh

My cousin thought I was booting up DOS from 1990s on my latest laptop when she was that wall of text 😂.
Even though she was extremely impressed with Gnome later on.
